### PR TITLE
fix(no delivery files check)

### PR DESCRIPTION
**Pull request by ChrOertlin**
## Description

Moves the no delivery files check to the delivery file fetcher.
adds customer inbox path to delivery meta data, this was needed to fetch the inbox path from the model instead of sample files. As with the new changes empty sample files were allowed, this would have crashed.

### Added

-

### Changed

-

### Fixed

-

### How to prepare for test

- [ ] Ssh to relevant server (depending on type of change)
- [ ] Use stage: `us`
- [ ] Paxa the environment: `paxa`
- [ ] Install on stage (example for Hasta):
    ```shell
    bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_cg -t cg -b [THIS-BRANCH-NAME] -a
    ```

### How to test

- [ ] Do ...

### Expected test outcome

- [ ] Check that ...
- [ ] Take a screenshot and attach or copy/paste the output.

## Review

- [ ] Tests executed by
- [ ] "Merge and deploy" approved by
Thanks for filling in who performed the code review and the test!

### This [version](https://semver.org/) is a

- [ ] **MAJOR** - when you make incompatible API changes
- [ ] **MINOR** - when you add functionality in a backwards compatible manner
- [ ] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

## Implementation Plan

- [ ] Document in ...
- [ ] Deploy this branch on ...
- [ ] Inform to ...

### CODE REVIEW

File: cg/services/deliver_files/delivery_file_fetcher_service/delivery_file_fetcher_service.py [hunk 1]
```diff
@@ -28,3 +32,13 @@ def __init__(
     def get_files_to_deliver(self, case_id: str) -> DeliveryFiles:
         """Get the files to deliver."""
         pass
+
+    @staticmethod
+    def _validate_delivery_has_content(delivery_files: DeliveryFiles) -> DeliveryFiles:
```
Comment (islean) [inline cg/services/deliver_files/delivery_file_fetcher_service/delivery_file_fetcher_service.py:37]: Slight preference for having this method not return anything and just raising an error if all files are missing, but feel free to go either way

File: cg/services/deliver_files/delivery_file_fetcher_service/error_handling.py [hunk 2]
```diff
@@ -14,10 +15,20 @@ def handle_missing_bundle_errors(func):
...
             return func(*args, **kwargs)
-        except HousekeeperBundleVersionMissingError as error:
-            LOG.error(error)
+        except HousekeeperBundleVersionMissingError:
+            msg: str = f"Missing bundles detected for:"
+            if case_id:
+                msg += f" case {case_id}"
+            if sample_id:
+                msg += f" sample {sample_id}"
+            if case:
```
Comment (islean) [inline cg/services/deliver_files/delivery_file_fetcher_service/error_handling.py:29]: A bit confused by the possibility of both a case_id and a case here. But if not important, feel free to resolve.

**Review by islean — COMMENTED** [review]
I like it! But do we actually set the customer_inbox at any point?

File: cg/services/deliver_files/delivery_file_fetcher_service/delivery_file_fetcher_service.py (hunk 1 above)
Comment (ChrOertlin) [reply cg/services/deliver_files/delivery_file_fetcher_service/delivery_file_fetcher_service.py:37]: I understand, I will keep it this way then. The other one would require an extra line of code :P

**Comment by ChrOertlin** [comment]
# test

ticket with cases that are not in HK
image

The messages seem a lot, but this is a ticket where nothing is in HK. So now all messages are written, this is not the case when just one sample is missing in a case for example

**Comment by islean** [comment]
> I like it! But do we actually set the customer_inbox at any point?

Just pinging this question since it was the bulk of the review

**Comment by diitaz93** [comment]
In `cg/cli/deliver/utils.py:deliver_raw_data_for_analyses` we will set the uploaded_at date to `datetime.now()` even if there are no files to deliver. This is the current behaviour, (so not a consequence of the code in this PR) but I still think it should not be that way. This happens because when there are no files the function `deliver_files_for_case` exits successfully and returns None. The same happens for all the pipeline's UploadAPI method `upload`.

File: cg/services/deliver_files/delivery_file_fetcher_service/error_handling.py (hunk 2 above)
Comment (ChrOertlin) [reply cg/services/deliver_files/delivery_file_fetcher_service/error_handling.py:29]: two decorators that get called in the appropraite place

**Comment by ChrOertlin** [comment]
> > I like it! But do we actually set the customer_inbox at any point?
> 
> Just pinging this question since it was the bulk of the review

yes it is set in the move service

**Review by islean — APPROVED** [review]
💯

File: cg/services/deliver_files/delivery_file_fetcher_service/error_handling.py [hunk 3]
```diff
@@ -14,10 +15,20 @@ def handle_missing_bundle_errors(func):
...
+                msg += f" case {case.internal_id}"
+            LOG.error(msg)
```
Comment (diitaz93) [inline cg/services/deliver_files/delivery_file_fetcher_service/error_handling.py:31]: 💪

**Review by diitaz93 — APPROVED** [review]
Looks good! Still concerned about the uploaded at date updated if there are no files to upload, maybe it is worth taking a look

File: cg/services/deliver_files/delivery_file_fetcher_service/error_handling.py [hunk 4]
```diff
@@ -14,10 +15,20 @@ def handle_missing_bundle_errors(func):
...
     def wrapper(*args, **kwargs):
+        case_id: str = kwargs.get("case_id")
+        sample_id: str = kwargs.get("sample_id")
+        case: Case = kwargs.get("case")
         try:
             return func(*args, **kwargs)
-        except HousekeeperBundleVersionMissingError as error:
-            LOG.error(error)
+        except HousekeeperBundleVersionMissingError:
+            msg: str = f"Missing bundles detected for:"
```
Comment (diitaz93) [inline cg/services/deliver_files/delivery_file_fetcher_service/error_handling.py:24]:
```suggestion
            msg: str = "Missing bundles detected for:"
```

**Pull request merged**